.circleci/config.yml

```diff
@@ -92,385 +108,894 @@ commands:
+      - checkout
+      - restore-kfp-cache
+      - run:
+          name: Install system deps
+          command: sudo apt update && sudo apt-get install -y libsndfile1 ffmpeg
+      - run:
+          name: Use py37
+          command: |
+            pyenv install -s 3.7.0
+            pyenv versions
+            pyenv global 3.7.0
+      - run:
+          name: Install python dependencies
+          command: |
+            pip install --upgrade pip
+            pip install tox==<< pipeline.parameters.tox_version >>
+      - run:
+          name: Install kubectl
+          command: |
+            curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+            sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+      - run:
+          name: Install kind
+          command: |
+            curl -Lo ./kind/kind https://kind.sigs.k8s.io/dl/v0.11.1/kind-linux-amd64 --create-dirs
+            chmod +x ./kind/kind
+            export PATH="./kind:$PATH"
+            kind create cluster
+      - run:
+          name: Install KFP
+          command: |
+            export PIPELINE_VERSION=1.7.1
+            kubectl apply -k "github.com/kubeflow/pipelines/manifests/kustomize/cluster-scoped-resources?ref=$PIPELINE_VERSION"
+            sleep 10s
+            kubectl wait --for condition=established --timeout=60s crd/applications.app.k8s.io
+            kubectl apply -k "github.com/kubeflow/pipelines/manifests/kustomize/env/platform-agnostic-pns?ref=$PIPELINE_VERSION"
+            sleep 10s
+            kubectl wait --for condition=ready --timeout=300s pods -n kubeflow --all
+            kubectl port-forward -n kubeflow svc/ml-pipeline-ui 8080:80 &
+      - run:
+          name: Run tests
+          command: |
+            tox -v -e << parameters.toxenv >>
+          no_output_timeout: 25m
+      - save-kfp-cache
+      - save-test-results
+
+  slack_notify:
+    parameters:
+      message:
+        type: string
+        default: ":runner:"
+    docker:
+      - image: 'cimg/base:stable'
+    steps:
+      - slack/notify:
+          custom: |
+            {
+              "blocks": [
+                {
+                  "type": "section",
+                  "fields": [
+                    {
+                      "type": "plain_text",
+                      "text": "<< parameters.message >>", 
+                      "emoji": true
+                    }
+                  ]
+                }
+              ]
+            }
+          event: always
+          channel: $SLACK_SDK_NIGHTLY_CI_CHANNEL
+
+  # Build the docker image for a yea shard of the nightly workflow.
+  sdk_docker:
+    parameters:
+      description:
+        type: string
+      path:
+        type: string
+      image_name:
+        type: string
+      python_version:
+        type: string
+        default: "3.8"
+      git_branch:
+        type: string
+        default: "nightly"
+#        default: "master"
```

Comment:
commented out

Reply:
yea, I'll swap "nightly" for "master" right before this goes in